libs/net/dal/Services/UserService.cs

```diff
@@ -146,5 +146,18 @@ public override User UpdateAndSave(User entity)
 
         throw new InvalidOperationException("User does not exist");
     }
+
+    public IEnumerable<User> FindByRoles(IEnumerable<string> roles)
+    {
+        var query = Context.Users.AsNoTracking();
+        var result = roles.Any() ? Enumerable.Empty<User>() : query;
+
+        foreach (var role in roles)
+        {
+            result = result.UnionBy(query.Where(c => EF.Functions.Like(c.Roles.ToLower(), $"%[{role.ToLower()}]%")), x => x.Id);
```

Comment:
You will get duplicates with this query.  Rewrite `result.UnionBy` with `result.Where` to support an Or statement

Reply:
Updated, thanks